core/src/main/java/org/geysermc/geyser/util/InventoryUtils.java

```diff
@@ -455,7 +455,7 @@ public static GeyserRecipe getValidRecipe(final GeyserSession session, final @Nu
                             for (int col = firstCol; col < width + firstCol; col++) {
                                 GeyserItemStack geyserItemStack = inventoryGetter.apply(col + (row * gridDimensions) + 1);
                                 if (geyserItemStack.isEmpty()) {
-                                    inventoryHasItem = itemStack == null || itemStack.getId() == 0;
+                                    inventoryHasItem = itemStack.getId() == 0;
                                     if (inventoryHasItem) {
```

Comment:
This change seems sus - maybe ask Camo if they remember this code

Reply:
![image](https://github.com/GeyserMC/Geyser/assets/105284508/a54dab5a-6076-4173-a5b3-b2803455bcd6)
`ingredient.getOptions()` is NonNull annotated